samza-kafka/src/test/scala/org/apache/samza/checkpoint/kafka/TestKafkaCheckpointManager.scala

```diff
@@ -69,23 +68,39 @@ class TestKafkaCheckpointManager extends KafkaServerTestHarness {
 
   val systemStreamPartitionGrouperFactoryString = classOf[GroupByPartitionFactory].getCanonicalName
 
+  var systemConsumerF: ()=>SystemConsumer = ()=>{null}
+  var systemProducerF: ()=>SystemProducer = ()=>{null}
+  var systemAdminF: ()=>SystemAdmin = ()=>{null}
+
   @Before
   override def setUp {
     super.setUp
 
     TestUtils.waitUntilTrue(() => servers.head.metadataCache.getAliveBrokers.size == numBrokers, "Wait for cache to update")
 
-    val config = new java.util.HashMap[String, Object]()
+    val systemName = "kafka"
     val brokers = brokerList.split(",").map(p => "localhost" + p).mkString(",")
-
+    val config = new java.util.HashMap[String, String]()
     config.put(ProducerConfig.BOOTSTRAP_SERVERS_CONFIG, brokers)
     config.put("acks", "all")
     config.put(ProducerConfig.MAX_IN_FLIGHT_REQUESTS_PER_CONNECTION, "1")
     config.put(ProducerConfig.RETRIES_CONFIG, (new Integer(java.lang.Integer.MAX_VALUE-1)).toString)
     config.putAll(KafkaCheckpointManagerFactory.INJECTED_PRODUCER_PROPERTIES.asJava)
-    producerConfig = new KafkaProducerConfig("kafka", "i001", config)
+    producerConfig = new KafkaProducerConfig(systemName, "i001", config)
 
     metadataStore = new ClientUtilTopicMetadataStore(brokers, "some-job-name")
+
+    config.put(SystemConfig.SYSTEM_FACTORY format systemName, "org.apache.samza.system.kafka.KafkaSystemFactory")
+    config.put(org.apache.samza.config.KafkaConfig.CHECKPOINT_SYSTEM, systemName);
+    config.put(JobConfig.JOB_NAME, "some-job-name");
+    config.put(JobConfig.JOB_ID, "i001");
+    config.put("systems.%s.producer.%s" format (systemName, ProducerConfig.BOOTSTRAP_SERVERS_CONFIG), brokers)
+    config.put("systems.%s.consumer.zookeeper.connect" format systemName, zkConnect)
+    val cfg: SystemConfig = new SystemConfig(new MapConfig(config))
+    val (sName: String, systemConsumerFactory : SystemFactory) =  KafkaCheckpointManagerFactory.getCheckpointSystemStreamAndFactory(cfg)
```

Comment:
s/sName/systemName

Reply:
there is already  a systemName variable above. Changed to something else.